src/vector_index.c

```diff
@@ -128,7 +128,7 @@ void VectorFilter_Free(VectorFilter *vf) {
     case VECSIM_QT_INVALID:
       break;
   }
-  for (int i = 0; i < vf->paramNum; i++) {
+  for (int i = 0; i < array_len(vf->params); i++) {
```

Comment:
Could use this [macro](https://github.com/RediSearch/RediSearch/blob/82e2cb286880e7f831e683b1bb1020a4128c6b88/src/query_param.h#L34)

Could even be shortened?
From
`#define QueryParam_NumParams(p) ((p)->params ? array_len((p)->params) : 0)`
To
`#define QueryParam_NumParams(p) (array_len((p)->params))`

Reply:
The macro can be shortened as `array_len()` test for null input be itself. 
Using this macro will work only because I used the same name here, it is not the same list (but I **can** use it)

Reply:
OK, different `params`. No need.